### ISSUE TITLE
Recognition error with emoji

### ISSUE THREAD

**Issue by yuanlinssdut** [Microsoft/vscode#20254]
- VSCode Version:1.9.0
- OS Version:macOs 10.12.3

![image](https://cloud.githubusercontent.com/assets/7608992/22773402/f3731ffc-eedc-11e6-8612-4ccc5035551f.png)

![biu](https://cloud.githubusercontent.com/assets/7608992/22773581/e5421996-eedd-11e6-887b-87b9398042ac.gif)

**Comment by Huddo121** [comment]
I've come across the same issue recently.

macOS Sierra 10.12.3
VS Code 1.9.1

I've had this issue turn up in Elixir files, but to rule out extensions being the problem I tried the same thing in a HTML file and observed the same behaviour.

![elixir_editing_optimized](https://cloud.githubusercontent.com/assets/1518405/23325556/60abaa34-fb4a-11e6-92e3-af7dd42bc03f.gif)
![html_editing_optimized](https://cloud.githubusercontent.com/assets/1518405/23325557/60afa594-fb4a-11e6-94be-8a1497a32b5a.gif)

This seems to be some sort of issue with multibyte unicode chars, as the behaviour is observed on every third press of the space bar in the above GIFs. The emoji that appears as two ? marks occurs at position 48 in the string when prepending with spaces, but position 24 in the string when prepending with even more beers. These question marks are still treated as a single character.

In IEX, a bit of playing around reveals the following
````
String.codepoints("🍺")      
["🍺"]

String.length("🍺")    
1

byte_size("🍺")    
4
````

Hope this helps 👍

**Comment by alexandrudima** [comment]
This has already been fixed in 1.10 stable via 72fe8f51b7ab3cec581d9ad1c6ffdd0be3a9349f for #20624
